### ISSUE TITLE
clang::TranslationUnit::parse should return Option<TranslationUnit>

### ISSUE THREAD

**Issue by fitzgen** [servo/rust-bindgen#144]
The `parse` method does an FFI call to `clang_parseTranslationUnit` which returns the "null" translation unit if parsing failed. We don't currently check that result, forcing callers to remember to check it, which is a foot gun. Instead, we should check it in this method and return `None` if parsing failed, or `Some` if it succeeded.

I can mentor whoever would like to work on this.

**Comment by igilham** [comment]
I'd like to have a go at this but could use some help as I've never written Rust before.

I know how to return `Option<TranslationUnit>` but don't really know how to handle it in calling classes. The docs for Option are good, but I could use a little help understanding the semantics of the caller. It looks like the caller currently assumes the pointer is not null and just runs with it (see src/ir/context.rs:106, 685).

I can imagine either throwing some sort of error on receipt of `None` or stuffing the `Option` into the struct that currently contains the `TranslationUnit` then handling it in the various places it is used. Is the `BindgenContext` in a valid state if it contains a `None` value and we keep executing?

**Comment by fitzgen** [comment]
Hi!

This would involve changing the function signature from `-> TranslationUnit` to `-> Option<TranslationUnit>`.

As for callers, you are right that they currently assume that they get back a valid translation unit. We can let them keep that expectation, and safely+loudly panic when that expectation is not met. We could do that by calling either `.unwrap()` or `.expect("diagnostic message blah blah")` on the returned `Option` value to get the underlying `Some` value, or panicking when the `Option` is a `None`.

Does that make sense?

**Comment by igilham** [comment]
I think so. does this look about right?

https://github.com/igilham/rust-bindgen/commit/362eac2b12568428637b7cdf091db3fbc9d46c00

**Comment by fitzgen** [comment]
> I think so. does this look about right?

Yeah! That looks great! Does it compile and do the tests pass? If so, sounds like it's time to open a PR! If not, I can help figure out what's going on, if needed.

**Comment by igilham** [comment]
Hmm... looks like the tests didn't pass to begin with (on this machine anyway). Here's the log of the previous master commit before my changes.

I'm running on OS X 10.10.5 with LLVM 3.9.

Does this seem familiar?

```
Running target/debug/tests-f239832665bb002e

running 1 test
test run_bindgen_tests ... FAILED

failures:

---- run_bindgen_tests stdout ----
	FAIL: /Users/gilhai01/code/rust-bindgen/tests/headers/jsval_layout_opaque.hpp
child stdout> run-bindgen.py: running ['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/jsval_layout_opaque.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '--', '-std=c++11', '/Users/gilhai01/code/rust-bindgen/tests/headers/jsval_layout_opaque.hpp']
child stderr> run-bindgen.py: running ['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/jsval_layout_opaque.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '--', '-std=c++11', '/Users/gilhai01/code/rust-bindgen/tests/headers/jsval_layout_opaque.hpp']
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroExpansion (502)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: MacroDefinition (501)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: StaticAssert (602)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: StaticAssert (602)
child stderr> ERROR:bindgen::ir::ty: invalid type Type(, kind: Invalid, decl: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None), canon: Cursor( kind: NoDeclFound, loc: builtin definitions, usr: None))
child stderr> ERROR:bindgen::ir::item: Unhandled cursor kind: StaticAssert (602)
child stderr> Traceback (most recent call last):
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 190, in <module>
child stderr>     main()
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 184, in main
child stderr>     generate_bindings(args.bindgen, test_flags, args.header, args.rust_bindings)
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 137, in generate_bindings
child stderr>     run_cmd(command, cwd=os.getcwd(), env=make_bindgen_env())
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 130, in run_cmd
child stderr>     subprocess.check_call(command, **kwargs)
child stderr>   File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 541, in check_call
child stderr>     raise CalledProcessError(retcode, cmd)
child stderr> subprocess.CalledProcessError: Command '['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/jsval_layout_opaque.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '--', '-std=c++11', '/Users/gilhai01/code/rust-bindgen/tests/headers/jsval_layout_opaque.hpp']' returned non-zero exit status -11
FAIL: /Users/gilhai01/code/rust-bindgen/tests/headers/union_with_anon_struct_bitfield.h
child stdout> run-bindgen.py: running ['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/union_with_anon_struct_bitfield.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '/Users/gilhai01/code/rust-bindgen/tests/headers/union_with_anon_struct_bitfield.h']
child stderr> run-bindgen.py: running ['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/union_with_anon_struct_bitfield.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '/Users/gilhai01/code/rust-bindgen/tests/headers/union_with_anon_struct_bitfield.h']
child stderr> Traceback (most recent call last):
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 190, in <module>
child stderr>     main()
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 184, in main
child stderr>     generate_bindings(args.bindgen, test_flags, args.header, args.rust_bindings)
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 137, in generate_bindings
child stderr>     run_cmd(command, cwd=os.getcwd(), env=make_bindgen_env())
child stderr>   File "/Users/gilhai01/code/rust-bindgen/tests/tools/run-bindgen.py", line 130, in run_cmd
child stderr>     subprocess.check_call(command, **kwargs)
child stderr>   File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 541, in check_call
child stderr>     raise CalledProcessError(retcode, cmd)
child stderr> subprocess.CalledProcessError: Command '['/Users/gilhai01/code/rust-bindgen/target/debug/bindgen', '-o', '/Users/gilhai01/code/rust-bindgen/tests/expectations/union_with_anon_struct_bitfield.rs', '--no-unstable-rust', '--raw-line', '', '--raw-line', '#![allow(non_snake_case)]', '--raw-line', '', '--no-unstable-rust', '/Users/gilhai01/code/rust-bindgen/tests/headers/union_with_anon_struct_bitfield.h']' returned non-zero exit status -11
thread 'run_bindgen_tests' panicked at '2 test failures!', tests/tests.rs:132
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    run_bindgen_tests

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured

error: test failed
```

**Comment by fitzgen** [comment]
Hmmm... The tests pass on master for me on OSX 10.11.6 and on Fedora 24, both with llvm 3.9.

Can you open a PR so we can see what Travis CI says and I can more easily pull down your changes locally?

**Comment by igilham** [comment]
Sure, I'll give it a try. It may just be an environment configuration issue.

**Comment by fitzgen** [comment]
Are running the tests with `DYLD_LIBRARY_PATH` and `LIBCLANG_PATH` set?

**Comment by igilham** [comment]
I can check that tomorrow. I left that machine at the office.

**Comment by igilham** [comment]
Travis seems to like it. I'll see if I can figure out the test environment when the opportunity arises.

**Comment by fitzgen** [comment]
Fixed in #153
